src/pages/integrations/gift-cards/confirm-card-purchase/confirm-card-purchase.ts

```diff
@@ -175,11 +175,15 @@ export class ConfirmCardPurchasePage extends ConfirmPage {
     this.externalLinkProvider.open(url);
   }
 
-  private resetValues() {
+  public resetValues() {
     this.totalAmountStr = this.invoiceFee = this.networkFee = this.totalAmount = this.wallet = null;
     this.tx = this.message = this.invoiceId = null;
   }
 
+  async publishInvoiceAndSign(tx, wallet) {
```

Comment:
exposes ConfirmPage.publishAndSign() to ConfirmInvoicePage because ConfirmPurchasePage overwrites publishAndSign so this.publishAndSign in ConfirmInvoicePage will use the incorrect overwritten method.